components/gitpod-cli/cmd/top.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	supervisor_helper "github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor-helper"
+	supervisor "github.com/gitpod-io/gitpod/supervisor/api"
+
+	"github.com/spf13/cobra"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+func outputTable(workspaceResources *supervisor.ResourcesStatusResponse) {
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"CPU (millicores)", "Memory (bytes)"})
+	table.SetBorders(tablewriter.Border{Left: true, Top: true, Right: true, Bottom: false})
+	table.SetCenterSeparator("|")
+
+	cpuFraction := int64((float64(workspaceResources.Cpu.Used) / float64(workspaceResources.Cpu.Limit)) * 100)
+	memFraction := int64((float64(workspaceResources.Memory.Used) / float64(workspaceResources.Memory.Limit)) * 100)
+	cpu := fmt.Sprintf("%dm/%dm (%d%%)", workspaceResources.Cpu.Used, workspaceResources.Cpu.Limit, cpuFraction)
+	memory := fmt.Sprintf("%dMi/%dMi (%d%%)\n", workspaceResources.Memory.Used/(1024*1024), workspaceResources.Memory.Limit/(1024*1024), memFraction)
+
+	cpuColor := getColor(cpuFraction)
+	memoryColor := getColor(memFraction)
+
+	table.Rich([]string{cpu, memory}, []tablewriter.Colors{{cpuColor}, {memoryColor}})
+
+	table.Render()
+}
+
+func getColor(value int64) int {
```

Comment:
I went for a simple "semaphore" approach (green / yellow / red).

Threshold are randomly picked based on common sense - happy to adjust them :)

Reply:
Can we provide a flag to disable coloring? I tried `watch gp top` and it does not look nice for some reason.

Reply:
I am implementing an improvement to detect if `gp` is being called by another process or is piped with stdin/stdout and disable colors accordingly.

I will also provide the flag for completeness 👍 

PR moved to draft while I finish this change.

Reply:
I've updated the PR, now `watch gp top` should just work (colors will be automatically disabled)